src/FiniteElement.m

```diff
@@ -982,9 +983,12 @@
                                 end
                             end
                             if (obj.fe_idx ~= 1 || ~settings.no_initial_impacts)
-                                cross_comp_pairs{2,jj+2} = [obj.w(obj.ind_Y_gap{1}), obj.w(obj.ind_lambda_normal{jj,1})]; % TODO maybe we need the other cross comps?
-                                len_pad = size(cross_comp_pairs{1,jj+2},1) - size(cross_comp_pairs{2,jj+2}, 1);
-                                cross_comp_pairs{2,jj+2} = vertcat(cross_comp_pairs{2,jj+2}, zeros(len_pad, 2));
+                                comp = [obj.w(obj.ind_Y_gap{1}), obj.w(obj.ind_lambda_normal{jj,1})]; % TODO maybe we need the other cross comps?
+                                size_comp = size(comp, 1);
+                                size_full_comp = size(cross_comp_pairs{end,jj+2});
+                                sparsity = Sparsity.rowcol([0:(size_comp-1)],[0,1],size_full_comp(1),size_full_comp(2));
+                                cross_comp_pairs{2,jj+2} = define_casadi_symbolic(settings.casadi_symbolic_mode, 'dummy', size_full_comp, sparsity);
```

Comment:
I am a bit confused why you need a matrix here. Can you explain?

Reply:
Because Sparsity based assignment doesn't work the way I would want to (i.e I don't think there is a way to change the sparsity of an existing matrix) so we need the matrix of the correct size and populate it afterward.